java/spacewalk-java.spec

```diff
@@ -53,97 +51,151 @@ BuildRoot:      %{_tmppath}/%{name}-%{version}-build
+%if 0%{?suse_version}
+BuildRequires:  ehcache >= 2.10.1
+%else
+BuildRequires:  ehcache-core
+%endif
+BuildRequires:  google-gson >= 2.2.4
+BuildRequires:  hibernate-commons-annotations
+BuildRequires:  hibernate5
+BuildRequires:  httpcomponents-asyncclient
+BuildRequires:  httpcomponents-client
+BuildRequires:  jade4j
+BuildRequires:  jaf
 %if 0%{?suse_version}
+BuildRequires:  jakarta-commons-discovery
+BuildRequires:  jakarta-commons-el
+BuildRequires:  jakarta-commons-fileupload
+BuildRequires:  jakarta-commons-validator
+%else
+BuildRequires:  apache-commons-discovery
+BuildRequires:  apache-commons-el
+BuildRequires:  apache-commons-fileupload
+BuildRequires:  apache-commons-validator
+%endif
+BuildRequires:  java-devel >= 11
+BuildRequires:  java-saml
+BuildRequires:  javapackages-tools
+BuildRequires:  javassist
+BuildRequires:  jboss-logging
+BuildRequires:  jcommon
+BuildRequires:  jdom
+BuildRequires:  joda-time
+BuildRequires:  jose4j
+BuildRequires:  jpa-api
+BuildRequires:  jsch
+BuildRequires:  jta
+BuildRequires:  libxml2
+BuildRequires:  libxml2-tools
+BuildRequires:  log4j
+BuildRequires:  netty
+BuildRequires:  objectweb-asm
+BuildRequires:  perl
+BuildRequires:  pgjdbc-ng
+BuildRequires:  postgresql-jdbc
+BuildRequires:  prometheus-client-java
+BuildRequires:  quartz
+BuildRequires:  redstone-xmlrpc
+BuildRequires:  salt-netapi-client >= 0.15.0
+BuildRequires:  simple-core
+BuildRequires:  simple-xml
+BuildRequires:  sitemesh
+BuildRequires:  snakeyaml
+BuildRequires:  spark-core
+BuildRequires:  spark-template-jade
+BuildRequires:  statistics
+BuildRequires:  stringtree-json
+BuildRequires:  struts >= 1.2.9
+BuildRequires:  tomcat >= 7
+BuildRequires:  tomcat-lib >= 7
+BuildRequires:  tomcat-taglibs-standard
+BuildRequires:  uyuni-base-server
+BuildRequires:  velocity
+BuildRequires:  woodstox
+BuildRequires:  xmlsec
+
+Requires:       /sbin/unix2_chkpwd
 Requires:       apache-commons-beanutils
+Requires:       apache-commons-cli
+Requires:       apache-commons-codec
 Requires:       apache-commons-collections
+Requires:       apache-commons-io
 Requires:       apache-commons-lang3
+Requires:       apache-commons-logging
+Requires:       bcel
+Requires:       byte-buddy
+Requires:       c3p0 >= 0.9.1
+Requires:       cglib
+%if 0%{?suse_version}
+Requires:       classmate
+Requires:       classpathx-mail
+%endif
 Requires:       cobbler >= 3.0.0
 Requires:       concurrent
-Requires:       google-gson >= 2.2.4
-Requires:       httpcomponents-client
-Requires:       jakarta-commons-digester
-Requires:       java >= 11
-Requires:       classmate
+Requires:       dwr >= 3
 Requires:       ehcache >= 2.10.1
 Requires:       gnu-jaf
-Requires:       byte-buddy
-Requires:       jpa-api
+Requires:       google-gson >= 2.2.4
 Requires:       hibernate-commons-annotations
 Requires:       hibernate5
+Requires:       httpcomponents-client
 Requires:       jade4j
+Requires:       jakarta-commons-digester
+Requires:       java >= 11
```

Comment:
For RH/Fedora, this needs to be `java >= 1:11` (for stupid historical reasons related to JPackage, there's _still_ an epoch...)

Also, wouldn't `java-headless` be better here? Do we _need_ the full Java stack with GUI libs?

(I see this requirement show up all over, I'm inquiring once to make it less annoying...)

Reply:
> Also, wouldn't java-headless be better here? Do we need the full Java stack with GUI libs?

Good question, but I would like to move this discussion to a different PR.